thor/tests/e2e.rs

```diff
@@ -5,397 +5,334 @@ mod harness;
 use anyhow::Result;
 use bitcoin::Amount;
 use bitcoin_harness::{self, Bitcoind};
+use futures::future;
 use genawaiter::GeneratorState;
-use harness::{build_runtime, generate_balances, make_transports, make_wallets, Transport};
+use harness::{generate_balances, make_transports, make_wallets, Transport, Wallet};
+use spectral::prelude::*;
+use testcontainers::clients::Cli;
 use thor::{Balance, Channel, PtlcPoint, PtlcSecret};
 
-#[test]
-fn e2e_channel_creation() {
-    let mut runtime = build_runtime();
-
-    let tc_client = testcontainers::clients::Cli::default();
-    let bitcoind = Bitcoind::new(&tc_client, "0.19.1").unwrap();
-    runtime.block_on(bitcoind.init(5)).unwrap();
-
-    let fund_amount_alice = Amount::ONE_BTC;
-    let fund_amount_bob = Amount::ONE_BTC;
-    let (balance_alice, balance_bob) = generate_balances(fund_amount_alice, fund_amount_bob);
-
+// NOTE: For some reason running these tests overflows the stack. In order to
+// mitigate this run them with:
+//
+//     RUST_MIN_STACK=10000000 cargo test
+
+async fn create_channels(
+    bitcoind: &Bitcoind<'_>,
+) -> (
+    Amount,
+    Channel,
+    Channel,
+    Transport,
+    Transport,
+    Wallet,
+    Wallet,
+    u32,
+    Amount,
```

Comment:
```suggestion
    Amount,
    (Channel,
    Transport, 
    Wallet),
    (Channel,
    Transport, 
    Wallet),
    u32,
    Amount,
```

Maybe a tuple is not appropriate, or maybe we should have a struct for `(Channel, Transport, Wallet)`, but I expect these element to be grouped per actor.

Reply:
Yeah I thought of adding a struct for this, while doing so the best name I could come up with was `Foo` - this made me leave it as it is :) I'll have another play with it, thanks.

Reply:
On further investigation I remember now why I didn't add the struct (besides being named `Foo`), the mutability requirements are different and I didn't want to have the whole struct mutable when only some fields needed to be. Not sure if there is a workaround for that?

Reply:
Leave it at tuples then. It's not so bad when reading the call site (which I hadn't done yet when writing the original comment).